### PR TITLE
Some bugfixes

### PR DESCRIPTION
**Pull request by dreibh**
Bugfixes for:
- https://github.com/NEAT-project/neat/issues/235
- https://github.com/NEAT-project/neat/issues/237
- https://github.com/NEAT-project/neat/issues/231

### CODE REVIEW

File: neat_core.c [hunk 1]
```diff
@@ -4083,7 +4083,7 @@ neat_write_to_lower_layer(struct neat_ctx *ctx, struct neat_flow *flow,
 
         msghdr.msg_flags = 0;
         if (flow->socket->fd != -1) {
-            rv = sendmsg(flow->socket->fd, (const struct msghdr *)&msghdr, 0);
+            rv = sendmsg(flow->socket->fd, (const struct msghdr *)&msghdr, MSG_NOSIGNAL);
```
Comment (bagder) [inline neat_core.c]: `MSG_NOSIGNAL` is not universally available. It is probably wise to use an #ifdef or something...

**Comment by bagder** [comment]
So this work replaces #233 ?

**Comment by dreibh** [comment]
Yes, the pull request includes the fix for #233.

**Comment by dreibh** [comment]
Added check for MSG_NOSIGNAL.

**Comment by bagder** [comment]
BTW: If you want `MSG_NOSIGNAL` behavior on Mac/BSD as well, you have to set the `SO_NOSIGPIPE` setsockopt() on the socket.

**Comment by weinrank** [comment]
@dreibh 
Please create a branch with your changes in this repository.
This will trigger the buildbots for configure, compile and runtime tests.

**Comment by dreibh** [comment]
Hi,

Fredag 23. desember 2016 00.58.49 skrev Felix Weinrank:
> @dreibh 
> Please create a branch with your changes in this repository.
> This will trigger the buildbots for configure, compile and runtime tests.

The branch is already there, since this is a pull request: dreibh/some-fixes.

-- 
Best regards / Mit freundlichen Grüßen / Med vennlig hilsen

=======================================================================
 Thomas Dreibholz

 Simula Research Laboratory
 Simula Innovation AS, Network Systems Group
 Visiting address: Martin Linges vei 17, 1364 Fornebu, Norway
 Mailing address:  P.O.Box 134, 1325 Lysaker, Norway
-----------------------------------------------------------------------
 E-Mail:     dreibh@simula.no
 Homepage:   http://simula.no/people/dreibh
=======================================================================

**Pull request merged**